plumbing/object/commit.go

```diff
@@ -215,6 +243,21 @@ func (b *Commit) Encode(o plumbing.EncodedObject) error {
 		return err
 	}
 
+	if b.GPGSignature != "" {
+		if _, err = fmt.Fprint(w, "gpgsig"); err != nil {
```

Comment:
Maybe we should try to mimic the original format of the commit content here (i.e. don't print this gpgsig field that didn't exist in the original commit, just print the signature at the end of the commit comment).

If we do leave this in we should probably rename to `pgpsig` anyways.

Reply:
Looking at `git cat-file` of a commit with signature, I don't think printing the signature at the end of commit comment would be good. The signature is not needed all the time but only when we want to verify. So, I would say we leave it with its own field. And yes, `pgpsig`, which doesn't matters much because we are using the header and footer of signature to detect them 😊